src/components/atoms/Heading/Heading.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { FC, HTMLAttributes } from 'react';
+import styled from 'styled-components';
+import Text from '../Text/Text';
+import { typography } from '../../../constants/typography';
+
+const {
+  sizes: { heading: headingSizes },
+} = typography;
+
+interface IStyledHeadingProps extends HTMLAttributes<HTMLHeadingElement> {
+  as: 'h1' | 'h2' | 'h3' | 'h4';
+}
+
+const StyledHeading = styled(Text)<IStyledHeadingProps>`
+  font-size: ${({ as = 'h3' }) => headingSizes[as]};
```

Comment:
I'm not sure if you need to give heading a default. When you write regular HTML, you still have to pick between h1 and h6